src/scripts/rhsm/download_file_and_report_progress.py

```diff
@@ -0,0 +1,36 @@
+import requests
```

Comment:
I want the download to report progress, and in that I need the request to stream=True in order to print out after specific-sized chunks. Otherwise the program would just wait on the `requests.get` and then print the progress on 100%.
I could not find ability to stream thru standart urllib library. But maybe I'm just missing something, correct me if I'm wrong.

Reply:
This is fairly straightforward:
```py
import urllib.request

with urllib.request.urlopen("https://github.com/cockpit-project/cockpit/releases/download/272/cockpit-272.tar.xz") as f:
    size = int(f.headers["content-length"])
    read = 0
    while block := f.read(2 ** 20):
        read += len(block)
        print(f"got {read} out of {size} bytes")
```

Of course you need to actually do something with the read blocks, like open a destination file and writing them there.